matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerExporter.java

```diff
@@ -129,33 +135,147 @@ static class Context {
 
         String refBusId;
 
-        int num = 1;
+        int num = 0;
 
         final Map<String, Integer> mBusesNumbersByIds = new HashMap<>();
 
         final List<String> generatorIdsConvertedToLoad = new ArrayList<>();
 
+        final Set<Bus> busesOutsideMainSynchronousComponentToBeConsidered = new HashSet<>();
+
         public Context(double maxGeneratorActivePowerLimit, double maxGeneratorReactivePowerLimit) {
             this.maxGeneratorActivePowerLimit = maxGeneratorActivePowerLimit;
             this.maxGeneratorReactivePowerLimit = maxGeneratorReactivePowerLimit;
         }
+
+        private void obtainBusesOutsideMainSynchronousComponentToBeConsidered(Network network) {
```

Comment:
You should better explain why you are doing all this complex topology processing. I personnally didn't understand without additionnal explanations. We have to be aware that all this processing to get synhcronous components exclusing LCC is technical debt for the code. If there is no other way to support HVDC export, why not but at least we have to document it.